tests/time_tests/test_runner/conftest.py

```diff
@@ -133,22 +133,43 @@ def temp_dir(pytestconfig):
 
 
 @pytest.fixture(scope="function")
-def cl_cache_dir(pytestconfig):
+def cl_cache_dir(pytestconfig, instance):
     """Generate directory to save OpenCL cache before test run and clean up after run.
 
     Folder `cl_cache` should be created in a directory where tests were run. In this case
     cache will be saved correctly. This behaviour is OS independent.
     More: https://github.com/intel/compute-runtime/blob/master/opencl/doc/FAQ.md#how-can-cl_cache-be-enabled
     """
-    cl_cache_dir = pytestconfig.invocation_dir / "cl_cache"
-    # if cl_cache generation to a local `cl_cache` folder doesn't work, specify
-    # `cl_cache_dir` environment variable in an attempt to fix it (Linux specific)
-    os.environ["cl_cache_dir"] = str(cl_cache_dir)
-    if cl_cache_dir.exists():
+    if instance["device"]["name"] == "GPU":
+        cl_cache_dir = pytestconfig.invocation_dir / "cl_cache"
+        # if cl_cache generation to a local `cl_cache` folder doesn't work, specify
+        # `cl_cache_dir` environment variable in an attempt to fix it (Linux specific)
+        os.environ["cl_cache_dir"] = str(cl_cache_dir)
```

Comment:
I thought os.environ has no effect on the subprocess unless we explicitly pass it there. And if we do so would it be better to carry over env addition and apply it explicitly only to the command we need?

Reply:
@asomsiko it has effect, because we aren't specify any environment for `subprocess` run, so it inherit environment of the parent process (details in https://docs.python.org/3/library/subprocess.html#subprocess.run)